src/engine/renderer/glsl_source/liquid_fp.glsl

```diff
@@ -138,24 +137,17 @@ void	main()
 		color.rgb = mix(u_FogColor, color.rgb, fogFactor);
 	}
 
-	vec3 L;
-	vec3 ambCol;
-	vec3 lgtCol;
+	vec3 lightDir;
+	vec3 ambientColor;
+	vec3 lightColor;
 
 	ReadLightGrid( (var_Position - u_LightGridOrigin) * u_LightGridScale,
-		       L, ambCol, lgtCol );
-
-	// compute half angle in world space
-	vec3 H = normalize(L + viewDir);
+		       lightDir, ambientColor, lightColor );
 
-	// compute the light term
-	vec3 light = lgtCol * clamp(dot(normal, L), 0.0, 1.0);
+	vec4 diffuse = vec4(0.0, 0.0, 0.0, 1.0);
 
-#if defined(r_specularMapping)
 	// compute the specular term
-	vec3 specular = reflectColor * lgtCol * pow(clamp(dot(normal, H), 0.0, 1.0), u_SpecularExponent.x + u_SpecularExponent.y) * r_SpecularScale;
```

Comment:
I'm not really convinced that replacing 2 lines with a call to a big function with 4 ifdefs is really a simplification, but whatever.

Reply:
1. By reusing the _big function with 4 ifdefs_ we make sure the used specular code is actually used and tested and maintained. This to prevent the code to break without being noticed.
2. I spent a lot of time to be sure the original code did the same. Now that we know it does the same, the best way to keep that information and to make sure it will be always true is to reuse the same.

Also, this is a bit like reusing a boring wall-of-text free software license written by lawyers and writing a short custom license, people will trust more the boring license because it is widely used and it was already successfully defended in courts.

Reply:
Also, there is hope to merge liquid shader with lightmapping shader one day, so the surface would be displayed with the diffuse and normal textures as fallback if liquidmapping is disabled. The current design paints nothing if liquidmapping is disabled. See #261.